tests/unit/test_value.cpp

```diff
@@ -203,6 +203,13 @@ TEST_F(ValueTest, TestArray) {
 	SIArray_Append(&arrOther, nestedArr);
 	otherHashCode = SIValue_HashCode(arrOther);
 	ASSERT_EQ(origHashCode, otherHashCode);
+
+	// Test logic for nested type-checking
+	bool contains_double = SIArray_ContainsType(arr, T_DOUBLE);
+	ASSERT_TRUE(contains_double);
+
+	bool contains_string = SIArray_ContainsType(arr, T_STRING);
+	ASSERT_FALSE(contains_string);
```

Comment:
Please add a test which checks the combination of types: `SIArray_ContainsType(arr, t0&t1)`